dali/kernels/gpu_utils.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_GPU_UTILS_H_
+#define DALI_KERNELS_GPU_UTILS_H_
+
+#include <cuda_runtime.h>
+
+namespace dali {
+namespace kernels {
+
+class DeviceGuard {
+ public:
+  explicit DeviceGuard(int new_device) {
+    cudaGetDevice(&original_device_);
```

Comment:
Don't we want to propagate the error form these calls?

Reply:
According to the docs, `cudaGetDevice` never fails.
cudaSetDevice can fail, in theory, but I can't really imagine a scenario when this would happen, at least in this context.